### PR TITLE
Add a V2 prow-bootstrap.py harness config.

### PR DESCRIPTION
**Pull request by Kashomon**
This is a re-write of the existing harness to use the prow-bootstrap mechanism.

### CODE REVIEW

File: testing/Makefile [hunk 1]
```diff
@@ -13,6 +13,7 @@
 # limitations under the License.
 #
```
Comment (amj) [inline testing/Makefile]: Can we get a file level comment saying what this is supposed to build?  Also, your header block here says "Copyright the Kubernetes Authors", which seems, uh, not right?

File: testing/bootstrap.sh [hunk 2]
```diff
@@ -1,11 +1,28 @@
...
 #!/bin/bash
 #
 # This script is used to bootstrap the [prow
 # jobs](https://github.com/kubernetes/test-infra/tree/master/prow), which is
 # the Kubernetes open source test-runner, rather like Jenkins. In other words,
 # this pulls the Minigo repo at some point (i.e., at the a pull request or
 # commit sha) and then runs the Minigo tests.
-set -e
+
+set -o errexit
```
Comment (amj) [inline testing/bootstrap.sh]: why?

Comment (Kashomon) [reply testing/bootstrap.sh]: For explicitness. Mostly copied from prow scripts, but I rather like it

File: testing/Makefile (hunk 1 above)
Comment (Kashomon) [reply testing/Makefile]: Done

File: testing/bootstrap.sh (hunk 2 above)
Comment (amj) [reply testing/bootstrap.sh]: ok.

File: README.md [hunk 3]
```diff
@@ -114,6 +116,16 @@ Running unit tests
 BOARD_SIZE=9 python3 -m unittest discover tests
 ```
 
+Automated Tests
+----------------
+
+Minigo uses the Kubernetes
+[Prow](https://github.com/kubernetes/test-infra/tree/master/prow) to
+automatically test PRs. You can see the status of our automated tests by
```
Comment (amj) [inline README.md]: Can you say in one sentence how it does this?  Like "Prow pulls commits into a hermetic environment managed by [where?] and runs our unittests, builds our containers, runs our integration tests, and checks our lint rules"..... or something :)

File: test.sh [hunk 4]
```diff
@@ -16,7 +16,34 @@
...
+#
+# NOTE! If this file changes/moves, please change
+# https://github.com/kubernetes/test-infra/blob/master/jobs/config.json
+
+# Ensure we're running from this directory to ensure PYTHONPATH is set
+# correctly.
+cd "$(dirname "$0")"
+
+lint_fail=0
+ls *.py | xargs pylint || {
```
Comment (amj) [inline test.sh]: this is different than autopep8?

File: testing/Makefile [hunk 5]
```diff
@@ -1,18 +1,21 @@
...
 # distributed under the License is distributed on an "AS IS" BASIS,
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-#
+
+# This file builds the images that are used by prow to run the Minigo tests.
+
 IMG = gcr.io/minigo-testing/minigo-prow-harness
+IMG_V2 = gcr.io/minigo-testing/minigo-prow-harness-v2
```
Comment (amj) [inline testing/Makefile]: v2?

File: README.md (hunk 3 above)
Comment (Kashomon) [reply README.md]: Done

File: test.sh (hunk 4 above)
Comment (Kashomon) [reply test.sh]: This is linting. I think autopep8 is just a formatter?

File: testing/Makefile (hunk 5 above)
Comment (Kashomon) [reply testing/Makefile]: Yeah, v1 was a dud.

**Pull request merged**